src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -755,11 +757,8 @@ bool QgsWFSProvider::setSubsetString( const QString &theSQL, bool updateFeatureC
   QString errorMsg;
   if ( !mShared->computeFilter( errorMsg ) )
     QgsMessageLog::logMessage( errorMsg, tr( "WFS" ) );
-  reloadData();
-  if ( updateFeatureCount )
-    featureCount();
```

Comment:
Hi @rouault 

While planing to integrate the featureCount "reset" in the `reloadProviderData` function, I noticed a behavior that is not consistent. While we have usually at this position a functionality resetting the feature counter, so it would be recalculated on the next call of `featureCount()` it recalculates it here. Means, it makes the opposite (a `featureCount()` call would not need to calculate again). 

To make it consistent with the other providers it would be needed to make a flag telling `featureCount` to recalculate or to remove this `featureCount()` call here and let it recalculate always on this provider on a `featureCount` call. I decided for the second one not to implement something that might be wrong. Do you approve that?